src/xcode/ENA/ENA/Source/Scenes/Home/HomeViewController.swift

```diff
@@ -170,88 +160,15 @@ final class HomeViewController: UIViewController {
 	}
 
 	func showExposureSubmission(with result: TestResult? = nil) {
-		present(
-			AppStoryboard.exposureSubmission.initiateInitial { coder in
-				ExposureSubmissionNavigationController(
-					coder: coder,
-					exposureSubmissionService: self.homeInteractor.exposureSubmissionService,
-					submissionDelegate: self,
-					testResult: result
-				)
-			},
-			animated: true
-		)
-	}
-
-	func showDeveloperMenu() {
-		present(
-			AppStoryboard.developerMenu.initiateInitial(),
-			animated: true,
-			completion: nil
-		)
-	}
-
-	func showInviteFriends() {
-		navigationController?.pushViewController(
-			FriendsInviteController.initiate(for: .inviteFriends),
-			animated: true
-		)
+		delegate?.showExposureSubmission(with: result)
```

Comment:
What is the reason behind having methods like this? I am not a fan of wrapping one method in another method and introducing 3 lines of code whether you could have just had one. 
I would just use `delegate?.showExposureSubmission(with: result)` where the previous method was called.
Additionally, this way of wrapping methods in another one makes it harder for people to understand the indirections. Again, if it were a method that really capsulates complex logic, having an extra method is perfectly acceptable. But this is a no-brainer line of code.

Reply:
The idea comes from theory of coordinator pattern. View Controllers shouldn't know about another View Controllers or coordinators, the only thing they just do is say that it completed the flow or done some action. So View Controller doesn't manage the flow itself, for View Controller doesn't matter who handle this, it just delegate it. 

And it's good thing, home vc doesn't know about exposure detection vc or settings vc, home vc just should say `detailButtonTapped`, `settingsActionRequested` or `finishSubmitFlow` by delegates or closures and another object (coordinator) will do logic of creating new vc/scene/module and routing it.

Reply:
This was wrapped because the method is called from 4 places in this view controller, and once from ```HomeInteractor```.  There are two reasons: First, so it could be a single point of call to the delegate method, and second, without this ```HomeInterator``` would need to know about the ```Coordinator``` and it shouldn't.